### ISSUE TITLE
Component Promotions

### ISSUE THREAD

**Issue by mannycarrera4** [Workday/canvas-kit#1395]
## 💡 Suggestion

Canvas Kit is wanting to promote components that have been in `labs` or `preview`. Some of these components are in a stable state to be promoted while others either need some work or there's future plans to promote them in a later version. This outline serves as a suggestion to which components should be promoted in `v7`.

Fixes: https://github.com/Workday/canvas-kit/issues/1194

#### Guide for promotion

✅   Good candidate for promotion

🚫   Should not be promoted

💁   Need more info / I'm not sure

**Components to be promoted:** `Box, Flex, Stack`
- [x] Box, Flex, Stack

## Preview Components

- **Breadcrumbs** 💁
    - Last updated: 29 days ago to add overflow and before that 5 months ago
    - **Should be promoted?**
        - I think the component can benefit from the changes that where introduced in tabs as well as list system. Again Alan would have more insight here
        - Open issues: https://github.com/Workday/canvas-kit/issues/1314, https://github.com/Workday/canvas-kit/issues/1070
        
- **Color Picker** 🚫
    - Last updated: 6 months ago to update tokens
    - **Should be promoted?**
        - This component will probably be replaced with en example
        - Maybe move this to PT Plat
        - I have a working PR with nicholas to assess navigation
        - A lot of accessibility concerns to address
        - Maybe update it to the current work in PR
        - Plan to update with latest list work
- **Inputs [FormField, Textarea, Text Input] (Rusty's Work)** 💁 / ✅
    - Last updated: 22 days ago to introduce the new API that rusty worked
    - **Should be promoted?**
        - I think we had questions about some of the api choices
        - Overall this will have to be updated when we tackle checkbox and radio inputs as well
        - Evaluate form field and `as`
        - Add fieldset component?
        - Needs more discussion
        - Spike cc: Team
- **Menu** 💁
    - Last updated: 4 months ago to update icon color
    - **Should be promoted?**
        - I think some changes to the list model need to be updated before this gets merged
        - I think this component is really close and the work done in tabs has set this up nicely to be promoted very soon
        - Open issues: https://github.com/Workday/canvas-kit/issues/1198, https://github.com/Workday/canvas-kit/issues/1070, https://github.com/Workday/canvas-kit/issues/808, https://github.com/Workday/canvas-kit/issues/632
- **Select** 💁 / ✅
    - Last updated: 6 to 7 months ago for token updates
    - **Should be promoted?**
        - Open issues: https://github.com/Workday/canvas-kit/issues/780, https://github.com/Workday/canvas-kit/issues/607,
        - I think this component is in a pretty solid state, it does have a few open issues but i think it can be promoted
        - Leave in preview, update one in main?
        - wait on combobox
        - v9? after work with in v8
        
- **Side Panel** 💁 / ✅
    - Last updated: 4 months to update docs
    - **Should be promoted?**
        - I think we were already discussing about promoting this
        - Open issues: https://github.com/Workday/canvas-kit/issues/909, https://github.com/Workday/canvas-kit/issues/55
        - Leave in preview and convert to compound?
        - Gig?
        - v8?
        
## Lab Components

- **Combobox** 💁
    - Last updated: 3 months ago to add auto scroll
    - **Should be Promoted?**
        - This component still needs to be converted to a compound component
        - It uses a text input, maybe this should use the text input from preview and be moved to preview after conversion?
        - Open issues: https://github.com/Workday/canvas-kit/issues/1174, https://github.com/Workday/canvas-kit/issues/1171, https://github.com/Workday/canvas-kit/issues/1122
            - **Questions**
                - I think before promoting we should: Convert to compound component, address open issues, and move to preview?
- **Box**  ✅
    - Last updated: 3 months ago to omit props
    - **Should be Promoted?**
        - Alan has more insight into this component, but we do use it ourselves and haven't found any breaking issues
        - A lot of teams could benefit from using this component and we've used to build out exmaples
        - I think this could be a good component to promote
        - Should be in one package
- **Drawer** 💁
    - Last updated: 6 - 8 months ago
    - **Should be Promoted?**
        - Although this component is used, i feel like a lot of this can be built with components like box and flex. We also have Side Panel.
        - Should we maybe consider this for soft deprecation?
        - Open issues: https://github.com/Workday/canvas-kit/issues/712
- **Header**  🚫
    - Last updated: 2 months for soft deprecation
    - **Should be Promoted?**
        - I believe this is being soft deprecated?
        - This component can be built up using our primitives
        - Maybe we add an example of how to this than it actually it being a component
        - Open issues: https://github.com/Workday/canvas-kit/issues/1392, https://github.com/Workday/canvas-kit/issues/763, https://github.com/Workday/canvas-kit/issues/314
- **Layout**  ✅
    - Flex
    - Stack
    - Last updated: 7 months ago
    - **Should be Promoted?**
        - Same as Box, i think this would be a great component to promoted but Alan might have more insight into these
        - Move box into this package?
- **Search Form** 💁
    - Last updated: 3 months ago to rename component
    - **Should be Promoted?**
        - The component says that it would be used in a context of a header and the header is being deprecated
        - This a nice to have component, again, maybe a component that fits in our examples instead?
        - There's no open issues for it and only one team uses this component
        
    
    ## Discussion points
    
    - Communicate what we're promoting
    - Figure out deprecation process
    - Discuss future work on promotion for other components

**Comment by anicholls** [comment]
Can we also make a note to put `Box`, `Flex`, and `Stack` in the same package @alanbsmith ? This drives me nuts haha

**Comment by mannycarrera4** [comment]
Yep this was brought up!

**Comment by myvuuu** [comment]
Steps to promote components:
- A script is used to complete the file migration for promotions
- Any changes like text updates will be manual

**Comment by myvuuu** [comment]
@mannycarrera4 to split ticket into individual component promotions (Breadcrumbs, Box, etc.)
